src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -85,9 +86,9 @@ public void Initialize(Http3StreamContext context)
 
         _context = context;
 
-        _errorCodeFeature = _context.ConnectionFeatures.Get<IProtocolErrorCodeFeature>()!;
-        _streamIdFeature = _context.ConnectionFeatures.Get<IStreamIdFeature>()!;
-        _streamAbortFeature = _context.ConnectionFeatures.Get<IStreamAbortFeature>()!;
+        _errorCodeFeature = _context.ConnectionFeatures.GetRequiredFeature<IProtocolErrorCodeFeature>();
```

Comment:
You found a bug, this `!` was wrong! Now we get to figure out why. I'll check where IProtocolErrorCodeFeature should be coming from and see why we thought it should have been there.

Reply:
Oh, its probably the test not setting up all the fields.
https://github.com/dotnet/aspnetcore/blob/4656f159492b1fb021992f52caf82e5bb5a66ee7/src/Servers/Kestrel/shared/test/TestContextFactory.cs#L189
Want to try filling those in with fake features, or do you need some help with that?

Reply:
Yes, @Tratcher I guess. Seems like recent tests failure are related to this only!